url70ed135da7_ralfbiedert_jcores_tgz-pJ8-sandbox_SimpleFilterTestJ8/src/net/jcores/jre/cores/commons/CommonUI.java

```diff
@@ -0,0 +1,150 @@
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR
+ * TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+ * EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ * 
+ */
+package net.jcores.jre.cores.commons;
+
+import java.awt.Color;
+import java.lang.reflect.InvocationTargetException;
+
+import javax.swing.SwingUtilities;
+import javax.swing.UIManager;
+import javax.swing.UnsupportedLookAndFeelException;
+
+import net.jcores.jre.CommonCore;
+import net.jcores.jre.interfaces.functions.F0;
+
+/**
+ * UI helper functions, e.g., EDT or Color related things. 
+ * 
+ * @author Ralf Biedert
+ * @since 1.0
+ * 
+ */
+public class CommonUI extends CommonNamespace {
+    /** The colors we need for heat maps (0 == cold, 255 == hot) */
+    private Color heatmapColors[] = new Color[256];
+
+    /**
+     * Creates a common ui object.
+     * 
+     * @param commonCore
+     */
+    public CommonUI(CommonCore commonCore) {
+        super(commonCore);
+
+        for (int i = 0; i < this.heatmapColors.length; i++) {
+            double hue = 0;
+            hue = ((float) i / (float) this.heatmapColors.length);
+            hue = (hue / 2) + 0.5;
+
+            this.heatmapColors[i] = Color.getHSBColor((float) hue, 0.85f, 1.0f);
+        }
+    }
+
+    /**
+     * Executes the given function in the Event Dispatch Thread (EDT) at some
+     * point in the future.
+     * 
+     * @since 1.0
+     * @param f0 The function to execute.
+     */
+    public void edt(final F0 f0) {
+        SwingUtilities.invokeLater(new Runnable() {
+            @Override
+            public void run() {
+                f0.f();
+            }
+        });
+    }
+
+    /**
+     * Sets the system look and feel for this application. Call this method at the very
+     * beginning of your application.
+     * 
+     * @since 1.0
+     */
+    public void systemLnF() {
+        try {
+            UIManager.setLookAndFeel(UIManager.getSystemLookAndFeelClassName());
+        } catch (ClassNotFoundException e) {
+            e.printStackTrace();
+        } catch (InstantiationException e) {
+            e.printStackTrace();
+        } catch (IllegalAccessException e) {
+            e.printStackTrace();
+        } catch (UnsupportedLookAndFeelException e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Executes the given function in the Event Dispatch Thread (EDT) now, waiting until
+     * the function was executed.
+     * 
+     * @since 1.0
+     * @param f0 The function to execute.
+     */
+    public void edtnow(final F0 f0) {
+        try {
+            SwingUtilities.invokeAndWait(new Runnable() {
+                @Override
+                public void run() {
+                    f0.f();
+                }
+            });
+        } catch (InterruptedException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)